lighty-applications/lighty-bgp-app/src/main/java/io/lighty/applications/bgp/app/Main.java

```diff
@@ -0,0 +1,131 @@
+            ExecutionException, TimeoutException {
+        final Stopwatch stopwatch = Stopwatch.createStarted();
+        LOG.info(".__  .__       .__     __              .__           ");
+        LOG.info("|  | |__| ____ |  |___/  |_ ___.__.    |__| ____     ");
+        LOG.info("|  | |  |/ ___\\|  |  \\   __<   |  |    |  |/  _ \\ ");
+        LOG.info("|  |_|  / /_/  >   Y  \\  |  \\___  |    |  (  <_> ) ");
+        LOG.info("|____/__\\___  /|___|  /__|  / ____| /\\ |__|\\____/ ");
+        LOG.info("        /_____/     \\/      \\/      \\/            ");
+        LOG.info("Starting BGP lighty.io application...");
+
+
+        final Set<YangModuleInfo> additionalModules =
+                Sets.newHashSet(Iterables.concat(BgpConfigUtils.ALL_BGP_MODELS, RestConfConfigUtils.YANG_MODELS));
+
+        controller = new LightyControllerBuilder()
+                .from(ControllerConfigUtils.getDefaultSingleNodeConfiguration(additionalModules))
+                .build();
+        startLightyModule(controller);
+
+        restconf = CommunityRestConfBuilder
+                .from(RestConfConfigUtils.getDefaultRestConfConfiguration(controller.getServices()))
+                .build();
+
+        startLightyModule(restconf);
+        bgpModule = new BgpModule(controller.getServices());
+        startLightyModule(bgpModule);
+
+        LOG.info("BGP lighty.io application started in {}", stopwatch.stop());
+    }
+
+    @SuppressWarnings({"checkstyle:illegalCatch"})
+    public synchronized void stop() {
+        LOG.info("Shutting down BGP application ...");
+        try {
+            if (restconf != null) {
+                restconf.shutdown().get();
+            }
+        } catch (Exception e) {
+            if (e instanceof InterruptedException) {
+                Thread.currentThread().interrupt();
+            }
+            LOG.error("Exception while shutting down RESTCONF:", e);
+        }
+
+        try {
+            if (bgpModule != null) {
+                bgpModule.shutdown().get();
+            }
+        } catch (Exception e) {
+            if (e instanceof InterruptedException) {
+                Thread.currentThread().interrupt();
+            }
+            LOG.error("Exception while shutting down BGPModule:", e);
+        }
+
+        try {
+            if (controller != null) {
+                controller.shutdown().get();
+            }
+        } catch (Exception e) {
+            if (e instanceof InterruptedException) {
+                Thread.currentThread().interrupt();
+            }
+            LOG.error("Exception while shutting down controller:", e);
+        }
+
+    }
+
+    private static void startLightyModule(final LightyModule lightyModule) throws InterruptedException,
+            ExecutionException, TimeoutException {
+        LOG.info("Starting lighty module {}", lightyModule.getClass().getName());
+        lightyModule.start().get(MODULE_STARTUP_WAIT_TIME, TimeUnit.MILLISECONDS);
```

Comment:
Please check if lightyModule is started correctly and not returning false.

Reply:
done, reworked the startup a little bit